sklearn/tree/_criterion.pyx

```diff
@@ -79,22 +79,22 @@ cdef class Criterion:
 
         pass
 
-    cdef void reset(self) nogil:
+    cdef int reset(self) nogil except -1:
```

Comment:
If these functions do not have real return values, I think "void ... except *" is most appropriate.